src/ScreenshotManager.vala

```diff
@@ -138,19 +138,40 @@ namespace Gala
 
 			yield;
 			selection_area.destroy ();
-			
+
 			yield wait_stage_repaint ();
 			selection_area.get_selection_rectangle (out x, out y, out width, out height);
 		}
 
+		public static void create_dir_if_missing (string path) {
+			File file = File.new_for_path (path);
+			if (!file.query_exists ()) {
+				file.make_directory_with_parents ();
```

Comment:
You have to explicitly forward the error that this can throw in the method.
So:
`public static void create_dir_if_missing (string path) throws Error {`

and:
```
try {
    file.make_directory_with_parents ();
} catch (Error e) {
    throw e;
}
```

Reply:
Ah... still so much to learn.